cli/azd/pkg/tools/azcli/user_profile.go

```diff
@@ -22,13 +23,17 @@ type UserProfileService struct {
 
 func NewUserProfileService(
 	credentialProvider auth.MultiTenantCredentialProvider,
-	clientOptionsBuilderFactory *azsdk.ClientOptionsBuilderFactory,
+	transporter policy.Transporter,
 	cloud *cloud.Cloud,
 ) *UserProfileService {
-	coreClientOptions := clientOptionsBuilderFactory.NewClientOptionsBuilder().
-		WithCloud(cloud.Configuration).
-		WithPerCallPolicy(azsdk.NewMsGraphCorrelationPolicy()).
-		BuildCoreClientOptions()
+	coreClientOptions := &azcore.ClientOptions{
```

Comment:
Looking at the diff now seems like it might be better if we just have this take `* azcore.ClientOptions` and then we don't have to construct this object and can just have the DI system take care of that.  I'll try that out.